src/components/list/list.tsx

```diff
@@ -151,10 +151,14 @@ export class List {
         const listItems = this.items.filter(this.isListItem);
 
         if (!this.multiple) {
-            this.mdcList.selectedIndex = listItems.findIndex(
+            const selectedIndex = listItems.findIndex(
                 (item: ListItem) => item.selected
             );
 
+            if (selectedIndex !== -1) {
+                this.mdcList.selectedIndex = selectedIndex;
```

Comment:
I'm thinking we might want to deselect or set the selectedIndex to 0 or something if `selectedIndex === -1`, but I'll have to test the PR to see what actually happens first… That'll have to be another day though, hopefully tomorrow.

Reply:
Yes deselect sounds like an option, not sure how to do that though. I guess that if we set selectedIndex to 0, the first item would be selected (visually as well?), so that is probably not the right thing to do.
But I'm not 100% into this so yes its best if you test it out, thanks :)

Reply:
Can you add an example to that reproduces what you are doing, so that the bug is reproduced if this code change is reverted? So I know that I'm testing the correct thing.

Reply:
It's just for testing, so don't worry about it looking nice in the docs or anything.

Reply:
Really stupid example but something like this:

```
    @Prop()
    public methods: ListItem[];

    public componentWillLoad() {
        this.methods = [{ text: 'foo', value: 1 }];
    }

    public render() {
        return (
            <limel-list
                onChange={() => {
                    this.methods = [{ text: 'foo', value: 1 }];
                }}
                items={this.methods}
                type="radio"
            />
        );
    }
```

We are doing something similar in eSign (reassigning methods list), but in a more complex situation.

Reply:
Ok, I'll use that. Thanks 👍

Question though: Is the intention to deselect the selected item? If not, can't you make sure the selected item has the property `selected: true`? 🤔

Reply:
If I recall correctly, yes that is the intention. We are clearing the selected item when you close the select signing method modal in eSign.
I don't know whether that is bad UX or not 😄

Reply:
Or that might be wrong ⬆️ 
This happens when we are signing the document. When we do that, we will rerender everything because we will receive a new signing object. A consequence of that is that the methods will also be reseted because they are extracted from the signing object.
As you see, a little bit more complex situation. Would probably be possible to solve in eSign as well, but it feels like we want to support these situations?

Reply:
Yes, ideally, if the items are updated, and there is no longer one that is selected, we would like to deselect all items. The problem is that this fix doesn't actually do that. It just circumvents the error by creating an inconsistent state, and I think that's worse.

Unfortunately, there doesn't seem to be a way to deselect all items in MDC List, so if that's what you wish to achieve, I would suggest using the `key` attribute on limel-list to force the creation of a new list component every time the selection should be reset. Alternatively, if the selection shouldn't be reset, check which item was selected before you replace the current items, then select the corresponding item in the new set. If there's a chance the new set doesn't include the same item, you would still need to handle recreating the list when no item should be selected.

The `key` attribute is described at the end of the [_Conditionals_ section](https://stenciljs.com/docs/templating-jsx#conditionals) of the Stencil docs.

Reply:
Or, well… I guess an argument could be made that limel-list should take care of recreating the mdc-component if the items are reset and there is no longer a selected item… 🤔

I think that would be reasonable solution, but honestly, if you need a fix urgently, I suggest just fixing it in your add-on for now. You can always remove that once we've had the time to fix it in Lime Elements. This seems like an obscure enough use case that I don't think it's reasonable for us to give it enough prio to address it urgently.

Reply:
Yes I agree with you. Putting it in an inconsistent/wrong state doesn't sound like something we want to do now when you are explaining it!

I will close this, thanks for you valuable input. 
I will take a closer look on our case in cool down, maybe we can restructure a bit. I guess the downside with the `key` attribute is performance, but that is maybe negligible.

Reply:
Please feel free to reach out to me for help, if you need it. A fix in the add-on should be faster to get to production, as you really only need to consider the possible use cases within the context of the add-on, as opposed to the much more general stance that must be taken in Lime Elements.